rest-service/manager_rest/test/endpoints/test_plugins_updates.py

```diff
@@ -0,0 +1,365 @@
+            'hello_world').id
+        plugins_update = self._sm.get(models.PluginsUpdate, plugins_update_id)
+        plugins_update.state = STATES.SUCCESSFUL
+        self._sm.update(plugins_update)
+        with self.assertRaisesRegexp(
+                CloudifyClientError,
+                "Cannot finalize plugins update .+, it's not in the "
+                "{0} state\\.".format(STATES.EXECUTING_WORKFLOW)):
+            self.client.plugins_update.finalize_plugins_update(
+                plugins_update_id)
+
+    @suppress_plugins_change_detection
+    def test_finalize_updates_original_blueprint_plan(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update_id = self.client.plugins_update.update_plugins(
+            'hello_world').id
+        blueprint = self._sm.get(models.Blueprint, 'hello_world')
+        plan = deepcopy(blueprint.plan)
+        plan.inputs['dummy'] = 'hi'
+        blueprint.plan = plan
+        self._sm.update(blueprint)
+        # Sanity check
+        self.assertIn(
+            'dummy',
+            self._sm.get(models.Blueprint, 'hello_world').plan.inputs)
+        self.client.plugins_update.finalize_plugins_update(
+            plugins_update_id)
+        self.assertNotIn(
+            'dummy',
+            self._sm.get(models.Blueprint, 'hello_world').plan.inputs)
+
+    @suppress_plugins_change_detection
+    def test_finalize_updates_plugins_update_state(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update = self.client.plugins_update.update_plugins(
+            'hello_world')
+        self.assertNotEqual(plugins_update.state, STATES.SUCCESSFUL)
+        plugins_update = self.client.plugins_update.finalize_plugins_update(
+            plugins_update.id)
+        self.assertEqual(plugins_update.state, STATES.SUCCESSFUL)
+
+    @suppress_plugins_change_detection
+    def test_finalize_deletes_temp_blueprint(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update = self.client.plugins_update.update_plugins(
+            'hello_world')
+        self._sm.get(models.Blueprint, plugins_update.temp_blueprint_id)
+        plugins_update = self.client.plugins_update.finalize_plugins_update(
+            plugins_update.id)
+        with self.assertRaises(NotFoundError):
+            self._sm.get(models.Blueprint, plugins_update.temp_blueprint_id)
+
+
+random_resource_id = PluginsUpdateManager._get_available_random_resource_id
```

Comment:
why not import them and use "as" in the import ?

Reply:
The first one is a class method so it's not possible, regarding the second - I don't know what I was thinking. Good catch